### DISCUSSION TITLE
Xray with WireGuard/IKEv2 on two VPS

### DISCUSSION THREAD

**Discussion by TSpodeneyko** [XTLS/Xray-core#2530] [Q&A]
Hello!

I have two VPS in different countries:
On first (uncensored country) installed Xray and running XTLS Reality protocol.
On the other I want to run VPN-server and connect to my first VPS, because VPN-protocols need for my case.

I want this scheme (example for iOS):
Client`s IPhone -> Wireguard/IKEv2 Server with installed Xray Client -> Xray Server -> Ethernet

How can I befriend them?

**Comment by chika0801** [comment]
> Client`s IPhone -> Wireguard/IKEv2 Server with installed Xray Client -> Xray Server -> Ethernet

`Xray Server`

I understand based on your question that the location here is your 1st VPS. it should receive Wireguard protocol inbound (using Xray-core).

But in Xray-core the Wireguard protocol can only be used outbound. So for this location, you have to consider using another program to receive Wireguard inbound, and then forward it to a socks port on your local machine (127.0.0.1). At this time, Xray-core's inbound station listens to this socks port.

**Comment by computerscot** [comment]
It might be possible like this. I have no idea if this would actually work, and I have not tried it.

* **Client device**: Install and configure WireGuard client. Tell the client in its  WireGuard configuration that the Endpoint is the Relay server's IP address.
* **Relay server**: The supposed Endpoint of the first WireGuard peer turns out not to be WireGuard software after all. It is in fact the `dokodemo-door` input of an Xray client. This sends traffic over Xray to the final server.
* **Final server**: The Xray server accepts traffic from the Xray client on the Relay server. The "freedom" output is not in fact the open Internet but turns out to be the WireGuard Endpoint the client device is looking for.

**Comment by computerscot** [comment]
Ok now I tried it, and I think it is working. 

https://cscot.pages.dev/2023/09/08/wireguard-xray-relay/

**Reply by TSpodeneyko** [reply]
Oh, thank you very much!

Last question, why will this scheme not working?

+-------------------+
| Client device     |
|                   |
| WireGuard peer    |
|                   |
+---------+---------+
          |
          |
          V
+-------------------+
| Relay server      |
| Wireguard peer                  |
| Xray client       |
|                   |
+---------+---------+
          |
          |
          V
+-------------------+
| Final server      |
|                          |
| Xray server       |
+-------------------+

**Reply by computerscot** [reply]
I did not test that scenario.

**Reply by its0ka** [reply]
can you please report your performance? it's bad in my case and online games stutter
